components/tidb_query/src/rpn_expr/impl_arithmetic.rs

```diff
@@ -1253,4 +1274,90 @@ mod tests {
         s.parse().ok()
     }
+
+    #[test]
+    fn test_real_divide() {
+        let normal = vec![
+            (double2real(2.2), double2real(1.1), double2real(2.0)),
+            (double2real(2.33), double2real(-0.01), double2real(-233.0)),
+            (double2real(2.33), double2real(0.01), double2real(233.0)),
+            (None, double2real(2.0), None),
+            (double2real(123.0), None, None),
+        ];
+
+        for (lhs, rhs, expected) in normal {
+            let actual = RpnFnScalarEvaluator::new()
+                .push_param(lhs.clone())
+                .push_param(rhs.clone())
+                .evaluate(ScalarFuncSig::DivideReal)
+                .unwrap();
+
+            assert_eq!(actual, expected, "lhs={:?}, rhs={:?}", lhs, rhs);
+        }
+
+        let abnormal = vec![
+            (double2real(2.33), double2real(0.0)),
+            (double2real(2.33), double2real(-0.0)),
+        ];
+
+        // Vec<[(Flag, SqlMode, is_ok(bool), has_warning(bool))]>
+        let modes = vec![
+            // Warning
+            (Flag::empty(), SqlMode::empty(), true, true),
+            // Error
+            (
+                Flag::IN_UPDATE_OR_DELETE_STMT,
+                SqlMode::ERROR_FOR_DIVISION_BY_ZERO | SqlMode::STRICT_ALL_TABLES,
+                false,
+                false,
+            ),
+            // Ok
+            (
+                Flag::IN_UPDATE_OR_DELETE_STMT,
+                SqlMode::STRICT_ALL_TABLES,
+                true,
+                false,
+            ),
+            // Warning
+            (
+                Flag::IN_UPDATE_OR_DELETE_STMT | Flag::DIVIDED_BY_ZERO_AS_WARNING,
+                SqlMode::ERROR_FOR_DIVISION_BY_ZERO | SqlMode::STRICT_ALL_TABLES,
+                true,
+                true,
+            ),
+        ];
+
+        for (lhs, rhs) in abnormal {
+            for &(flag, sql_mode, is_ok, has_warning) in &modes {
+                // Construct an `EvalContext`
+                let mut config = EvalConfig::new();
+                config.set_flag(flag).set_sql_mode(sql_mode);
+
+                let (result, mut ctx) = RpnFnScalarEvaluator::new()
+                    .context(EvalContext::new(std::sync::Arc::new(config)))
+                    .push_param(lhs.clone())
+                    .push_param(rhs.clone())
+                    .evaluate_ctx::<Real>(ScalarFuncSig::DivideReal);
+
+                if is_ok {
+                    assert_eq!(result.unwrap(), None);
+                } else {
+                    assert!(result.is_err());
+                }
+
+                if has_warning {
+                    assert_eq!(
+                        ctx.take_warnings().warnings[0].get_code(),
+                        ERR_DIVISION_BY_ZERO
+                    );
+                } else {
+                    assert!(ctx.take_warnings().warnings.is_empty());
+                }
+            }
+        }
+    }
+
+    fn double2real(x: f64) -> Option<Real> {
```

Comment:
if you want to introduce this function, please also refine other places in this file to keep them consistent.